lib/src/webdav/client.dart

```diff
@@ -71,7 +97,7 @@ class WebDavClient {
     ];
     return _network.send(
       'MKCOL',
-      _getUrl(path),
+      _getUrl(await _addFilesPath(path)),
```

Comment:
Is there a reason for not merging `_getUrl` and `_addFilesPath` into one method?

Reply:
See line 80. Maybe merging the methods and using the url directly in that line makes more sense.

Reply:
You could also turn it around. Make `_addFilesPath` internally call `_getUrl`.

Reply:
Oh yeah that sounds good